src/data_snack/snack.py

```diff
@@ -119,42 +110,38 @@ def get_many(
         :param keys_values: list of keys, each list defines a set key values for one Entity object
         :return: a list of retrieved Entity objects
         """
-        type_name = cls.__name__
-        _keys = [
-            self.key_factory.get_key(type_name, *key_values)
-            for key_values in keys_values
-        ]
-        records = list(self.connection.get_many(cls, _keys).values())
-        return self._get_serializer(type_name).deserialize(records, many=True)
+        _keys = [self.key_factory(cls, key_values) for key_values in keys_values]
+        records = list(self.connection.get_many(_keys).values())
+        return self._get_serializer(cls).deserialize(records, many=True)
 
-    def set_many(self, entities: List[Entity]) -> List[Text]:
+    def set_many(self, entities: List[Entity]) -> Any:
         """
         Saves multiple `Entity` objects in db.
 
         :param entities: a list of Entity objects
         :return: a list of keys generated for saved objects
         """
-        cls = entities[0].__class__
-        type_name = cls.__name__
-        records = self._get_serializer(type_name).serialize(entities, many=True)
-        keys = [self._build_record_key(type_name, entity) for entity in entities]
-        if result := self.connection.set_many(cls, dict(zip(keys, records))):
+        # TODO: fail if entities is an empty list, which is still valid according
+        #  to the typing
+        entity_type = entities[0].__class__
```

Comment:
We have this magic `getattr(next(iter(` contraption in the connection. Would it be possible to reuse it here too?